frontend/src/main/pages/PlayPage.js

```diff
@@ -117,6 +118,30 @@ export default function PlayPage() {
     mutationsell.mutate(userCommons)
   };
 
+  const [isChatOpen, setIsChatOpen] = useState(false);
+
+  const toggleChatWindow = () => {
+    setIsChatOpen(!isChatOpen);
```

Comment:
Setting state based on current state will behave unexpectedly on batched updates. Pass in an updater callback instead.

Reply:
Not sure what this means? toggleChatWindow is the callback for the toggle chat button.

Reply:
`setIsChatOpen` doesn't change state immediately. State is always immutable, so it only changes on the next render. This means if any updates are batched, reading from state directly would mean reading from stale state. Instead pass in an updater callback when calculating new state based on previous state.
```suggestion
    setIsChatOpen((prevState) => !prevState);
```

Reply:
So then prevState is?

Reply:
This works, is this what you mean?

Reply:
My suggested change is all you need to change. setState accepts either the new state or a callback to calculate new state. The callback is needed when you need to calculate state based on previous state.

Reply:
Curious how this works? If you have the time

Reply:
Going to resolve this issue, but if you want to explain, please feel free in the PR comments

Reply:
Setting state doesn’t change the state immediately. It updates on the following render, meaning if react batches updates, you encounter situations where you’re reading the same stale state across multiple setState call.

In this case if setIsChatOpen gets called more than once, the resulting state would be either true or false depending on whether updates were batched. That’s bad and unpredictable.

Instead if you pass in an updater callback you can read the most up to date value of the state even on batched updates.